packages/fxa-auth-server/fxa-oauth-server/lib/jwt_access_token.js

```diff
@@ -14,20 +14,18 @@ const HEADER_TYP = 'at+JWT';
  */
 exports.create = async function generateJWTAccessToken(accessToken, grant) {
   const clientId = hex(grant.clientId);
+  // The IETF spec for `aud` refers to https://openid.net/specs/openid-connect-core-1_0.html#IDToken
+  // > REQUIRED. Audience(s) that this ID Token is intended for. It MUST contain the
+  // > OAuth 2.0 client_id of the Relying Party as an audience value. It MAY also contain
+  // > identifiers for other audiences. In the general case, the aud value is an array of
+  // > case sensitive strings. In the common special case when there is one audience, the
+  // > aud value MAY be a single case sensitive string.
+  const audience = grant.resource ? [clientId, grant.resource] : clientId;
```

Comment:
It's my continuing opinion that applying the `"It MUST contain the OAuth 2.0 client_id of the Relying Party as an audience value"` requirement from `id_token`s to `access_token`s is at best unnecessary, and at worst a harmful confusion of responsibilities that could lead to subtle security problems. But I can see how a strict read of both specs would lead to it being a requirement, so I guess I should take it up with the spec...

Reply:
I agree, it seems a bit strange. A couple of alternatives if no `resource` is specified:

* Don't send `aud` since it'll be the same as `client_id`.
* Send an empty array